api/STRUCTURE.md

```diff
@@ -0,0 +1,83 @@
+# All of Us Workbench API Structure & Technologies
```

Comment:
Please add an inbound link from the top level README (one directory above api). Actually, this could be merged with (or replace) some of the preamble material there.

Reply:
That's fair. I might rename this new one to something like `BACKEND_OVERVIEW.md` or something. (There are structures everywhere.)

That one is a train wreck. I have a couple of tickets stashed away on reorganizing it and breaking it into useful sub-docs.